crates/ln-dlc-node/src/tests/mod.rs

```diff
@@ -1,6 +1,368 @@
+/// transaction to the given `address`.
+async fn fund_and_mine(address: bitcoin::Address, amount: bitcoin::Amount) {
+    #[derive(Serialize)]
+    struct Payload {
+        address: bitcoin::Address,
+        #[serde(with = "bdk::bitcoin::util::amount::serde::as_btc")]
+        amount: bitcoin::Amount,
+    }
+
+    let client = reqwest::Client::new();
+
+    let result = client
+        .post(format!("{CHOPSTICKS_FAUCET_ORIGIN}/faucet"))
+        .json(&Payload { address, amount })
+        .send()
+        .await
+        .unwrap();
+
+    assert!(result.status().is_success());
+}
+
+/// Instructs `nigiri-chopsticks` to mine a block.
+async fn mine(n: u32) {
+    let address =
+        Address::from_str("bcrt1qylgu6ffkp3p0m8tw8kp4tt2dmdh755f4r5dq7s").expect("valid address");
+
+    for _ in 0..n {
+        fund_and_mine(address.clone(), bitcoin::Amount::from_sat(1000)).await;
+    }
+
+    // For the mined blocks to be picked up by the subsequent wallet
+    // syncs
+    tokio::time::sleep(Duration::from_secs(5)).await;
+}
+
+fn random_tmp_dir() -> PathBuf {
+    let tmp = temp_dir();
+
+    let rand_string = thread_rng()
+        .sample_iter(&Alphanumeric)
+        .take(10)
+        .map(char::from)
+        .collect::<String>();
+
+    let tmp = tmp.join(rand_string);
+
+    tracing::debug!(
+        path = %tmp.to_str().expect("to be a valid path"),
+        "Generated temporary directory string"
+    );
+
+    tmp
+}
+
+#[allow(dead_code)]
+fn log_channel_id(node: &Node, index: usize, pair: &str) {
+    let details = node
+        .channel_manager
+        .list_channels()
+        .get(index)
+        .unwrap()
+        .clone();
+
+    let channel_id = hex::encode(details.channel_id);
+    let short_channel_id = details.short_channel_id.unwrap();
+    let is_ready = details.is_channel_ready;
+    let is_usable = details.is_usable;
+    let inbound = details.inbound_capacity_msat;
+    let outbound = details.outbound_capacity_msat;
+    tracing::info!(
+        channel_id,
+        short_channel_id,
+        is_ready,
+        is_usable,
+        inbound,
+        outbound,
+        "{pair}"
+    );
+}
+
+async fn wait_until<P, T, F>(timeout: Duration, predicate_fn: P) -> Result<T>
+where
+    P: Fn() -> F,
+    F: Future<Output = Result<Option<T>>>,
+{
+    tokio::time::timeout(timeout, async {
+        loop {
+            match predicate_fn().await? {
+                Some(value) => return Ok(value),
+                None => tokio::time::sleep(Duration::from_millis(500)).await,
+            };
+        }
+    })
+    .await?
+}
+
+fn dummy_contract_input(
+    offer_collateral: u64,
+    accept_collateral: u64,
+    oracle_pk: XOnlyPublicKey,
+) -> ContractInput {
+    let total_collateral = offer_collateral + accept_collateral;
 
-#[cfg(feature = "system_tests")]
-mod system_tests;
+    ContractInput {
+        offer_collateral,
+        accept_collateral,
+        maturity_time: 1653378780,
```

Comment:
Should we dynamically compute this value to ensure it is always in the future?

Reply:
It doesn't seem to matter right now, but I've set it to `now + 24 hours` just in case it starts mattering.